### ISSUE TITLE
docker build fails at node stage

### ISSUE THREAD

**Issue by tmatth** [xiph/awcy#225]
To reproduce:
```
docker build .
```

<details><summary>Console output:</summary>

```
Step 27/31 : RUN 	cd ${APP_DIR} && 	export PYTHON=python2.7 && 	npm install && 	npm run tsc && 	cd ${APP_DIR}/www && 	npm install && 	npm run build
 ---> Running in b97ecbee4861
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @types/form-data@2.5.0: This is a stub types definition. form-data provides its own type definitions, so you do not need this installed.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

> iconv@2.2.3 install /opt/app/node_modules/iconv
> node-gyp rebuild

make: Entering directory '/opt/app/node_modules/iconv/build'
  CXX(target) Release/obj.target/iconv/src/binding.o
../src/binding.cc:35:11: error: ‘v8::Handle’ has not been declared
   35 | using v8::Handle;
      |           ^~~~~~
../src/binding.cc:65:26: error: ‘Handle’ has not been declared
   65 |   static void Initialize(Handle<Object> obj)
      |                          ^~~~~~
../src/binding.cc:65:32: error: expected ‘,’ or ‘...’ before ‘<’ token
   65 |   static void Initialize(Handle<Object> obj)
      |                                ^
../src/binding.cc: In static member function ‘static void {anonymous}::Iconv::Initialize(int)’:
../src/binding.cc:70:5: error: ‘obj’ was not declared in this scope
   70 |     obj->Set(Nan::New<String>("make").ToLocalChecked(),
      |     ^~~
../src/binding.cc:71:60: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’
   71 |              Nan::New<FunctionTemplate>(Make)->GetFunction());
      |                                                            ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note: candidate: ‘v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)’
 6126 |   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
      |                                              ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note:   candidate expects 1 argument, 0 provided
../src/binding.cc:73:63: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’
   73 |              Nan::New<FunctionTemplate>(Convert)->GetFunction());
      |                                                               ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note: candidate: ‘v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)’
 6126 |   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
      |                                              ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note:   candidate expects 1 argument, 0 provided
../src/binding.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE {anonymous}::Iconv::Make(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/binding.cc:84:44: error: no matching function for call to ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::Value>)’
   84 |     String::Utf8Value from_encoding(info[0]);
      |                                            ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3135:5: note: candidate: ‘v8::String::Utf8Value::Utf8Value(v8::Isolate*, v8::Local<v8::Value>)’
 3135 |     Utf8Value(Isolate* isolate, Local<v8::Value> obj);
      |     ^~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3135:5: note:   candidate expects 2 arguments, 1 provided
../src/binding.cc:85:42: error: no matching function for call to ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::Value>)’
   85 |     String::Utf8Value to_encoding(info[1]);
      |                                          ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3135:5: note: candidate: ‘v8::String::Utf8Value::Utf8Value(v8::Isolate*, v8::Local<v8::Value>)’
 3135 |     Utf8Value(Isolate* isolate, Local<v8::Value> obj);
      |     ^~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3135:5: note:   candidate expects 2 arguments, 1 provided
../src/binding.cc:92:64: error: no matching function for call to ‘v8::ObjectTemplate::NewInstance()’
   92 |         Nan::New<ObjectTemplate>(object_template)->NewInstance();
      |                                                                ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6423:44: note: candidate: ‘v8::MaybeLocal<v8::Object> v8::ObjectTemplate::NewInstance(v8::Local<v8::Context>)’
 6423 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(Local<Context> context);
      |                                            ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6423:44: note:   candidate expects 1 argument, 0 provided
../src/binding.cc: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE {anonymous}::Iconv::Convert(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/binding.cc:103:49: error: no matching function for call to ‘v8::Value::BooleanValue()’
  103 |     const bool is_flush = info[8]->BooleanValue();
      |                                                 ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2699:8: note: candidate: ‘bool v8::Value::BooleanValue(v8::Isolate*) const’
 2699 |   bool BooleanValue(Isolate* isolate) const;
      |        ^~~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2699:8: note:   candidate expects 1 argument, 0 provided
In file included from /root/.cache/node-gyp/12.16.1/include/node/v8-internal.h:14,
                 from /root/.cache/node-gyp/12.16.1/include/node/v8.h:27,
                 from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2702:51: note: candidate: ‘v8::Maybe<bool> v8::Value::BooleanValue(v8::Local<v8::Context>) const’
 2702 |                 V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
      |                                                   ^~~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2702:51: note:   candidate expects 1 argument, 0 provided
 2702 |                 V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
      |                                                   ^~~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
../src/binding.cc:106:47: error: no matching function for call to ‘v8::Value::Uint32Value()’
  106 |     size_t input_start = info[2]->Uint32Value();
      |                                               ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note: candidate: ‘v8::Maybe<unsigned int> v8::Value::Uint32Value(v8::Local<v8::Context>) const’
 2707 |   V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
      |                                         ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note:   candidate expects 1 argument, 0 provided
../src/binding.cc:107:46: error: no matching function for call to ‘v8::Value::Uint32Value()’
  107 |     size_t input_size = info[3]->Uint32Value();
      |                                              ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note: candidate: ‘v8::Maybe<unsigned int> v8::Value::Uint32Value(v8::Local<v8::Context>) const’
 2707 |   V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
      |                                         ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note:   candidate expects 1 argument, 0 provided
../src/binding.cc:109:48: error: no matching function for call to ‘v8::Value::Uint32Value()’
  109 |     size_t output_start = info[5]->Uint32Value();
      |                                                ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note: candidate: ‘v8::Maybe<unsigned int> v8::Value::Uint32Value(v8::Local<v8::Context>) const’
 2707 |   V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
      |                                         ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note:   candidate expects 1 argument, 0 provided
../src/binding.cc:110:47: error: no matching function for call to ‘v8::Value::Uint32Value()’
  110 |     size_t output_size = info[6]->Uint32Value();
      |                                               ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note: candidate: ‘v8::Maybe<unsigned int> v8::Value::Uint32Value(v8::Local<v8::Context>) const’
 2707 |   V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
      |                                         ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2707:41: note:   candidate expects 1 argument, 0 provided
../src/binding.cc:127:72: warning: ‘bool v8::Object::Set(uint32_t, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
  127 |     rc->Set(0, Nan::New<Integer>(static_cast<uint32_t>(input_consumed)));
      |                                                                        ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/v8-internal.h:14,
                 from /root/.cache/node-gyp/12.16.1/include/node/v8.h:27,
                 from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3508:22: note: declared here
 3508 |                 bool Set(uint32_t index, Local<Value> value));
      |                      ^~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
../src/binding.cc:128:73: warning: ‘bool v8::Object::Set(uint32_t, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
  128 |     rc->Set(1, Nan::New<Integer>(static_cast<uint32_t>(output_consumed)));
      |                                                                         ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/v8-internal.h:14,
                 from /root/.cache/node-gyp/12.16.1/include/node/v8.h:27,
                 from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../../nan/nan.h:56,
                 from ../src/binding.cc:18:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:3508:22: note: declared here
 3508 |                 bool Set(uint32_t index, Local<Value> value));
      |                      ^~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
make: *** [iconv.target.mk:118: Release/obj.target/iconv/src/binding.o] Error 1
make: Leaving directory '/opt/app/node_modules/iconv/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 5.4.0-59-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/app/node_modules/iconv
gyp ERR! node -v v12.16.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 

> node-icu-charset-detector@0.2.0 install /opt/app/node_modules/node-icu-charset-detector
> node-gyp rebuild

make: Entering directory '/opt/app/node_modules/node-icu-charset-detector/build'
  CXX(target) Release/obj.target/node-icu-charset-detector/node-icu-charset-detector.o
../node-icu-charset-detector.cpp:17:29: error: ‘Handle’ in namespace ‘v8’ does not name a template type
   17 |     RegisterClass(const v8::Handle<v8::Object> target) {
      |                             ^~~~~~
../node-icu-charset-detector.cpp:17:35: error: expected ‘,’ or ‘...’ before ‘<’ token
   17 |     RegisterClass(const v8::Handle<v8::Object> target) {
      |                                   ^
../node-icu-charset-detector.cpp:60:5: error: expected ‘;’ at end of member declaration
   60 |     FromBuffer(v8::Handle<v8::Object> bufferObject) {
      |     ^~~~~~~~~~
      |               ;
../node-icu-charset-detector.cpp:60:26: error: expected ‘)’ before ‘<’ token
   60 |     FromBuffer(v8::Handle<v8::Object> bufferObject) {
      |               ~          ^
      |                          )
../node-icu-charset-detector.cpp: In static member function ‘static void CharsetMatch::RegisterClass(int)’:
../node-icu-charset-detector.cpp:33:9: error: ‘target’ was not declared in this scope
   33 |         target->Set(Nan::New(ClassName).ToLocalChecked(), constructorTemplate->GetFunction());
      |         ^~~~~~
../node-icu-charset-detector.cpp:33:92: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’
   33 |         target->Set(Nan::New(ClassName).ToLocalChecked(), constructorTemplate->GetFunction());
      |                                                                                            ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../node-icu-charset-detector.cpp:1:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note: candidate: ‘v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)’
 6126 |   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
      |                                              ^~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:6126:46: note:   candidate expects 1 argument, 0 provided
../node-icu-charset-detector.cpp: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE CharsetMatch::New(Nan::NAN_METHOD_ARGS_TYPE)’:
../node-icu-charset-detector.cpp:72:13: error: ‘Handle’ is not a member of ‘v8’
   72 |         v8::Handle<v8::Object> buffer = info[0]->ToObject();
      |             ^~~~~~
../node-icu-charset-detector.cpp:72:30: error: expected primary-expression before ‘>’ token
   72 |         v8::Handle<v8::Object> buffer = info[0]->ToObject();
      |                              ^
../node-icu-charset-detector.cpp:72:32: error: ‘buffer’ was not declared in this scope; did you mean ‘setbuffer’?
   72 |         v8::Handle<v8::Object> buffer = info[0]->ToObject();
      |                                ^~~~~~
      |                                setbuffer
../node-icu-charset-detector.cpp:72:59: error: no matching function for call to ‘v8::Value::ToObject()’
   72 |         v8::Handle<v8::Object> buffer = info[0]->ToObject();
      |                                                           ^
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../node-icu-charset-detector.cpp:1:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2672:44: note: candidate: ‘v8::MaybeLocal<v8::Object> v8::Value::ToObject(v8::Local<v8::Context>) const’
 2672 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
      |                                            ^~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2672:44: note:   candidate expects 1 argument, 0 provided
In file included from /root/.cache/node-gyp/12.16.1/include/node/v8-internal.h:14,
                 from /root/.cache/node-gyp/12.16.1/include/node/v8.h:27,
                 from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../node-icu-charset-detector.cpp:1:
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2686:31: note: candidate: ‘v8::Local<v8::Object> v8::Value::ToObject(v8::Isolate*) const’
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h:2686:31: note:   candidate expects 1 argument, 0 provided
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
../node-icu-charset-detector.cpp:78:44: error: ‘CharsetMatch::FromBuffer’ cannot be used as a function
   78 |             CharsetMatch::FromBuffer(buffer)->Wrap(info.This()); // under GC
      |                                            ^
../node-icu-charset-detector.cpp: In function ‘void RegisterModule(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’:
../node-icu-charset-detector.cpp:125:33: error: cannot convert ‘Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE’ {aka ‘v8::Local<v8::Object>’} to ‘int’
  125 |     CharsetMatch::RegisterClass(target);
      |                                 ^~~~~~
      |                                 |
      |                                 Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE {aka v8::Local<v8::Object>}
../node-icu-charset-detector.cpp:17:19: note:   initializing argument 1 of ‘static void CharsetMatch::RegisterClass(int)’
   17 |     RegisterClass(const v8::Handle<v8::Object> target) {
      |                   ^~~~~~~~~~~~~~~~
In file included from ../node-icu-charset-detector.cpp:1:
../node-icu-charset-detector.cpp: At global scope:
/root/.cache/node-gyp/12.16.1/include/node/node.h:601:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
  601 |       (node::addon_register_func) (regfunc),                          \
      |                                           ^
/root/.cache/node-gyp/12.16.1/include/node/node.h:635:3: note: in expansion of macro ‘NODE_MODULE_X’
  635 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../node-icu-charset-detector.cpp:128:1: note: in expansion of macro ‘NODE_MODULE’
  128 | NODE_MODULE(node_icu_charset_detector, RegisterModule);
      | ^~~~~~~~~~~
In file included from /root/.cache/node-gyp/12.16.1/include/node/node.h:63,
                 from ../node-icu-charset-detector.cpp:1:
/root/.cache/node-gyp/12.16.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/root/.cache/node-gyp/12.16.1/include/node/node_object_wrap.h:85:78:   required from here
/root/.cache/node-gyp/12.16.1/include/node/v8.h:10226:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
10226 |                reinterpret_cast<Callback>(callback), type);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/.cache/node-gyp/12.16.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/root/.cache/node-gyp/12.16.1/include/node/v8.h:10226:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
make: *** [node-icu-charset-detector.target.mk:109: Release/obj.target/node-icu-charset-detector/node-icu-charset-detector.o] Error 1
make: Leaving directory '/opt/app/node_modules/node-icu-charset-detector/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 5.4.0-59-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/app/node_modules/node-icu-charset-detector
gyp ERR! node -v v12.16.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 

> sqlite3@4.2.0 install /opt/app/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download 
[sqlite3] Success: "/opt/app/node_modules/sqlite3/lib/binding/node-v72-linux-x64/node_sqlite3.node" is installed via remote
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: iconv@2.2.3 (node_modules/iconv):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: iconv@2.2.3 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: node-icu-charset-detector@0.2.0 (node_modules/node-icu-charset-detector):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: node-icu-charset-detector@0.2.0 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

added 193 packages from 241 contributors and audited 197 packages in 12.781s

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


> awcy@1.0.0 tsc /opt/app
> tsc

node_modules/@types/node/index.d.ts(20,1): error TS1084: Invalid 'reference' directive syntax.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! awcy@1.0.0 tsc: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the awcy@1.0.0 tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-01-06T05_22_47_083Z-debug.log
The command '/bin/sh -c cd ${APP_DIR} && 	export PYTHON=python2.7 && 	npm install && 	npm run tsc && 	cd ${APP_DIR}/www && 	npm install && 	npm run build' returned a non-zero code: 2
```
</details>

So it seems like `node-gyp` fails to rebuild iconv and tsc fails with `node_modules/@types/node/index.d.ts(20,1): error TS1084: Invalid 'reference' directive syntax.`

**Comment by tdaede** [comment]
At least the iconv failure doesn't seem to be fatal, the tsc error is though.

**Comment by tmatth** [comment]
Fixed in #264
